File: Programming Basics with JS/T04 - Nested Conditional Statements/exercise/04. Fishing Boat.js
function priceBoat(input) {
    let budget = Number(input[0]);
    let season = input[1];
    let countOfFishers = Number(input[2]);
    let price = 0;

    switch(season){
        case "Spring": price = 3000; break;
        case "Summer":
        case "Autumn": price = 4200; break;
        case "Winter": price = 2600; break;
    }

    if (countOfFishers <= 6) {
        price = price * 0.90;
    } else if (countOfFishers >= 7 && countOfFishers <= 11) {
        price = price * 0.85;
    } else {
        price = price * 0.75
    }

    if ((countOfFishers % 2 == 0) && season !== "Autumn") {
        price = price * 0.95;
    }
    let diff = Math.abs(price - budget);

    if (budget >= price) {
        console.log(`Yes! You have ${diff.toFixed(2)} leva left.`)
    } else {
        console.log(`Not enough money! You need ${diff.toFixed(2)} leva.`);
    }
}
priceBoat(["3000", "Summer", "11"])

File: Programming Basics with Python/T04 - Conditional Statements (Advanced)/exercise/04. Fishing Boat.py
budget = int(input())
season = str(input())
number_fishermen = int(input())
boat_price = 0
money_left = 0
money_short = 0

if season == "Spring":
    boat_price = 3000
    if number_fishermen <= 6:
        boat_price -= boat_price * 0.10
    elif 7 <= number_fishermen <= 11:
        boat_price -= boat_price * 0.15
    elif number_fishermen >= 12:
        boat_price -= boat_price * 0.25

elif season == "Summer" or season == "Autumn":
    boat_price = 4200
    if number_fishermen <= 6:
        boat_price -= boat_price * 0.10
    elif 7 <= number_fishermen <= 11:
        boat_price -= boat_price * 0.15
    elif number_fishermen >= 12:
        boat_price -= boat_price * 0.25

elif season == "Winter":
    boat_price = 2600
    if number_fishermen <= 6:
        boat_price -= boat_price * 0.10
    elif 7 <= number_fishermen <= 11:
        boat_price -= boat_price * 0.15
    elif number_fishermen >= 12:
        boat_price -= boat_price * 0.25

if number_fishermen % 2 == 0:
    if season == "Spring" or season == "Summer" or season == "Winter":
        boat_price -= boat_price * 0.05

if budget >= boat_price:
    money_left = budget - boat_price
    print(f"Yes! You have {money_left:.2f} leva left.")
elif budget < boat_price:
    money_short = boat_price - budget
    print(f"Not enough money! You need {money_short:.2f} leva.")